editor/plugins/canvas_item_editor_plugin.cpp

```diff
@@ -179,6 +179,10 @@ void CanvasItemEditor::_snap_if_closer_float(float p_value, float p_target_snap,
 	}
 }
 
+bool CanvasItemEditor::_is_node_editable(const Node *p_node) {
+	return (!p_node->has_meta("_edit_lock_") && !ClassDB::is_parent_class(p_node->get_parent()->get_class_name(), "Container"));
```

Comment:
You should check also here if the metadata is set to true using : `p_node->get_meta("_edit_lock_")`
Also, the check should test whether or not the child node inherits from the Control class, as other nodes are not affected by their parent container.

Reply:
@groud  Ah yes checking for `get_meta` confused me in some places it's done, in some not. Is there any reason for this?

Reply:
Not really, they are more or less equivalent I guess. But it is safer that way.

Reply:
Added those improvements!